test/images/agnhost/webhook/main.go

```diff
@@ -14,22 +14,46 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package main
+package webhook
 
 import (
 	"encoding/json"
-	"flag"
 	"fmt"
 	"io/ioutil"
 	"net/http"
 
+	"github.com/spf13/cobra"
+
 	"k8s.io/api/admission/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/klog"
 	// TODO: try this library to see if it generates correct json patch
 	// https://github.com/mattbaird/jsonpatch
 )
 
+var (
+	certFile string
+	keyFile  string
+)
+
+// CmdWebhook is used by agnhost Cobra.
+var CmdWebhook = &cobra.Command{
+	Use:   "webhook",
+	Short: "Starts a HTTP server, useful for testing MutatingAdmissionWebhook and ValidatingAdmissionWebhook",
+	Long: `Starts a HTTP server, useful for testing MutatingAdmissionWebhook and ValidatingAdmissionWebhook.
+After deploying it to Kubernetes cluster, the Administrator needs to create a ValidatingWebhookConfiguration
+in the Kubernetes cluster to register remote webhook admission controllers.`,
+	Args: cobra.MaximumNArgs(2),
```

Comment:
Similar to the nettest, it seems like the args here corresponds to the flag values; that isn't what the cobra `Args` are for though. It seems like both of those should be zero args exactly.

Reply:
Done